integration/test/test_power_balancer.py

```diff
@@ -0,0 +1,243 @@
+                    report_path = '{}.report'.format(run_name)
+                    trace_path = '{}.trace'.format(run_name)
+                    cls._tmp_files.append(report_path)
+                    cls._tmp_files.append(trace_path)
+                    launcher = geopm_test_launcher.TestLauncher(app_conf, agent_conf, report_path,
+                                                                trace_path, time_limit=2700)
+                    launcher.set_num_node(cls._num_node)
+                    launcher.set_num_rank(num_rank)
+                    launcher.write_log(run_name, 'Power cap = {}W'.format(power_budget))
+                    launcher.run(run_name)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean up any files that may have been created during the test if we
+        are not handling an exception and the GEOPM_KEEP_FILES
+        environment variable is unset.
+
+        """
+        if not cls._keep_files:
+            for path in cls._tmp_files:
+                for tf in glob.glob(path + '.*'):
+                    os.unlink(tf)
+
+    def tearDown(self):
+        if sys.exc_info() != (None, None, None):
+            TestIntegration_power_balancer._keep_files = True
+
+    def get_power_data(self, agent, report_path, trace_path):
+        output = geopmpy.io.AppOutput(report_path, trace_path + '*')
+        node_names = output.get_node_names()
+        self.assertEqual(self._num_node, len(node_names))
+
+        new_output = geopmpy.io.RawReport(report_path)
+        power_budget = new_output.meta_data()['Policy']['POWER_PACKAGE_LIMIT_TOTAL']
+
+        power_limits = []
+        # Total power consumed will be Socket(s) + DRAM
+        for nn in node_names:
+            tt = output.get_trace_data(node_name=nn)
+
+            first_epoch_index = tt.loc[tt['EPOCH_COUNT'] == 0][:1].index[0]
+            epoch_dropped_data = tt[first_epoch_index:]  # Drop all startup data
+
+            power_data = epoch_dropped_data[['TIME', 'ENERGY_PACKAGE', 'ENERGY_DRAM']]
+            power_data = power_data.diff().dropna()
+            power_data.rename(columns={'TIME': 'ELAPSED_TIME'}, inplace=True)
+            power_data = power_data.loc[(power_data != 0).all(axis=1)]  # Will drop any row that is all 0's
+
+            pkg_energy_cols = [s for s in power_data.keys() if 'ENERGY_PACKAGE' in s]
+            dram_energy_cols = [s for s in power_data.keys() if 'ENERGY_DRAM' in s]
+            power_data['SOCKET_POWER'] = power_data[pkg_energy_cols].sum(axis=1) / power_data['ELAPSED_TIME']
+            power_data['DRAM_POWER'] = power_data[dram_energy_cols].sum(axis=1) / power_data['ELAPSED_TIME']
+            power_data['COMBINED_POWER'] = power_data['SOCKET_POWER'] + power_data['DRAM_POWER']
+
+            pandas.set_option('display.width', 100)
+            # launcher.write_log(name, 'Power stats from {} {} :\n{}'.format(agent, nn, power_data.describe()))
```

Comment:
Maybe this should just be uncommented and left in all the time.  Having this data from the nightlies might be useful.  I think you could also wrap it in an if check for skip launch if you don't want to pollute the output log when you're running that way.

Reply:
It can be printed to stdout if you pass --show-details (on 186-187), but it won't be in the launcher log.  After splitting test into a launch step and analysis step, the launcher object doesn't actually exist in this function